lib/ansible/modules/windows/win_dns_record.ps1

```diff
@@ -149,12 +141,13 @@ if ($values -ne $null -and $values.Count -gt 0)
             $module.FailJson("Error adding DNS $type resource $name in zone $zone with value $value", $_)
         }
         $module.Result.debug_Add_DnsServerResourceRecord = "Add-DnsServerResourceRecord -ZoneName $zone -Name $name -AllowUpdateAny -TimeToLive $ttl -$type -$record_argument_name $value"
-        $changes += "+[$zone] $name $($ttl.TotalSeconds) $type $value`n"
     }
 
     $module.Result.changed = $true
 }
+$records_end = Get-DnsServerResourceRecord -ZoneName $zone -Name $name -RRType $type -Node -ErrorAction:Ignore @extra_args | Sort-Object
 
-$module.Diff.prepared = ($changes -Join '')
+$module.Diff.before = @($records | ForEach-Object { "[$zone] $($_.HostName) $($_.TimeToLive.TotalSeconds) $type $($_.RecordData.$record_argument_name.ToString())`n" }) -join ''
```

Comment:
These changes still don't work right in check mode -- because it should show what _would_ have been changed [(reference)](https://docs.ansible.com/ansible/2.7/user_guide/playbooks_checkmode.html#showing-differences-with-diff).

This diffing method may be elegant, but it's also demonstrably wrong.

Reply:
Ouch! 
I wanted to get rid of `prepared` (not appearing in [2.8 docs](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_general_windows.html#windows-new-module-development)) and didn't realize.

Reply:
Okay, that makes sense -- the lack of `prepared` seems a likely victim of the move away from the various legacy methods. That said, we'll still need to figure a way for diff-mode to be accurate when in check-mode. I'll take another stab at it today or tomorrow.